### PR TITLE
Added an option to save atom order when adding atom labels for a template reaction

### PR DESCRIPTION
**Pull request by alongd**
### Motivation or Problem
The Family.add_atom_labels_for_reaction() method may be used for 3D TS guess generation outside RMG. However, currently, it does not preserve the atom order of the original molecule (tracked down to a `struc.update()` call under `apply_recipe()`)

### Description of Changes
A `save_order` attribute was added to `KineticsFamily`.

### CODE REVIEW

**Comment by alongd** [comment]
@amarkpayne, could you take a look at this PR?

**Comment by amarkpayne** [comment]
Absolutely! Sorry, I forgot that I was assigned, but I'll get started now

File: rmgpy/reaction.pxd [hunk 1]
```diff
@@ -113,7 +114,9 @@ cdef class Reaction:
 
     cpdef bint can_tst(self) except -2
 
-    cpdef calculate_microcanonical_rate_coefficient(self, np.ndarray e_list, np.ndarray j_list, np.ndarray reac_dens_states, np.ndarray prod_dens_states=?, double T=?)
+    cpdef calculate_microcanonical_rate_coefficient(self, np.ndarray e_list, np.ndarray j_list,
+                                                    np.ndarray reac_dens_states, np.ndarray prod_dens_states=?,
+                                                    double T=?)
```
Comment (amarkpayne) [inline rmgpy/reaction.pxd:119]: This is just whitespace changes, right? If so can you move it to a separate minor commit?

File: rmgpy/data/kinetics/family.py [hunk 2]
```diff
@@ -1500,15 +1501,18 @@ def apply_recipe(self, reactant_structures, forward=True, unique=True):
...
+                struc.update()
             reactant_net_charge += struc.get_net_charge()
 
         for struct in product_structures:
             # If product structures are Molecule objects, update their atom types
             # If product structures are Group objects and the reaction is in certain families
             # (families with charged substances), the charge of structures will be updated
             if isinstance(struct, Molecule):
-                struct.update()
+                struct.update(sort_atoms=not self.save_order)
```
[the remark this answers is not in the document]
Comment (alongd) [reply rmgpy/data/kinetics/family.py:1515]: We would not like to sort the atoms if the user requested to save the atom order

**Comment by alongd** [comment]
Thanks for the comments! I addressed them in fixups (and rebased)

**Comment by alongd** [comment]
@amarkpayne, I addressed the initial comments, could you take a second look?

**Comment by alongd** [comment]
@amarkpayne, changes were added, a fixup commit (not uploaded previously) was squashed, another commit of style modifications to reaction.pxd was added.
Thanks!!

**Pull request merged**